pkg/attribute/mutableBag.go

```diff
@@ -302,55 +305,82 @@ func GetBagFromProto(attrs *mixerpb.Attributes, globalDict []string) (*MutableBa
 
 	// TODO: fail if the proto carries multiple attributes by the same name (but different types)
 
+	var buf *bytes.Buffer
+	if glog.V(2) {
+		buf = pool.GetBuffer()
+		log(buf, "Creating bag from wire attributes:\n")
+	}
+
+	log(buf, "  updating string attributes:\n")
 	for k, v := range attrs.Strings {
 		name, e = lookup(k, e, globalDict, messageDict)
 		value, e = lookup(v, e, globalDict, messageDict)
+		log(buf, "    %s -> '%s'\n", name, value)
 		mb.values[name] = value
 	}
 
+	log(buf, "  updating int64 attributes:\n")
 	for k, v := range attrs.Int64S {
 		name, e = lookup(k, e, globalDict, messageDict)
+		log(buf, "    %s -> '%d'\n", name, value)
 		mb.values[name] = v
 	}
 
+	log(buf, "  updating double attributes:\n")
```

Comment:
All these uses of "updating" are right. This is not a delta operation. "setting" would be better.